h2o-core/src/main/java/water/network/SSLSocketChannelFactory.java

```diff
@@ -0,0 +1,108 @@
+package water.network;
+
+import water.H2O;
+import water.util.Log;
+
+import javax.net.ssl.*;
+import java.io.FileInputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.channels.ByteChannel;
+import java.nio.channels.SocketChannel;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.util.Properties;
+
+public class SSLSocketChannelFactory {
+
+    private SSLContext sslContext = null;
+    private SSLProperties properties = null;
+
+    public SSLSocketChannelFactory() throws SSLContextException {
+        try {
+            SSLProperties props = new SSLProperties();
+            props.load(new FileInputStream(H2O.ARGS.ssl_config));
+            init(props);
+        } catch (IOException e) {
+            Log.err("Failed to initialized SSL context.", e);
+            throw new SSLContextException("Failed to initialized SSL context.", e);
+        }
+    }
+
+    public SSLSocketChannelFactory(SSLProperties props) throws SSLContextException {
+        init(props);
+    }
+
+    private void init(SSLProperties props) throws SSLContextException {
+        properties = props;
+        try {
+            if(sslParamsPresent()) {
+                this.sslContext = SSLContext.getInstance(properties.h2o_ssl_protocol());
+                this.sslContext.init(keyManager(), trustManager(), null);
+            } else {
+                this.sslContext = SSLContext.getDefault();
+            }
+        } catch (NoSuchAlgorithmException |
```

Comment:
ok can change to a chain of catch{} but are we supposed to be J6 compatible? Thought the Github page said we require 7+